### PR TITLE
Feature/refactor menu description

### PR DESCRIPTION
**Pull request by Schneegans**
This monster-pull-request addresses #37, #38, and #48. Here is the list of major changes:
* All predefined menus (`Bookmarks`, `Devices`, etc.) can now be top-level menus.
* There is no differentiation between top-level and submenus anymore. Therefore you can drag entire menus into other menus in the menu editor. Or you can make a former submenu to a top-level menu by drag'n'drop. The items `Top-Level Menu` and `Submenu` are now merged to a `Custom Menu`.
* The D-Bus interface of Fly-Pie now supports all action of menu types. You can open any menu you can configure in the menu editor also via the D-Bus interface. The README.md has been extended to contain a full description of the menu configuration format.
* The `ItemRegistry` has been refactor so that all action and menu types have their own files.

Due to all these changes, the translations need an update. Especially most mentions of "Submenu" will have to be replaced with "Menu".

### CODE REVIEW

**Comment by Schneegans** [comment]
@daPhipz Could you have a look at the updates to the German translation (https://github.com/Schneegans/Fly-Pie/pull/49/files#diff-6686c8173bf4fa70e9c9755259206f2c5c4fa9fde6269d36e5262dcc09abcf25)? 90 % of the diff are changed line numbers, but several strings have changed and some have been added.

It would be great if you could also check the new D-Bus section of the README.md (https://github.com/Schneegans/Fly-Pie/tree/feature/refactor-menu-description#fly-pies-d-bus-interface) for any typos.

Before I merge this PR, I will try to in-depth test all features of Fly-Pie. Especially the drag'n'drop features of the menu editor have become way more complex - I am sure that there are some bugs to be found! If you have some time I would greatly appreciate it if you could test this branch! Things we could test:

- [ ] Importing menu configurations from the `develop` branch.
- [ ] Exporting and re-importing a menu configuration in this branch.
- [ ] Functionality of all actions and menus.
- [ ] Functionality of menus when added as top-level menus.
- [ ] Internal drag'n'drop in the menu. With Live-Preview, with <kbd>Ctrl</kbd> pressed, from top-level to other menus, ... 
- [ ] External drag'n'drop. Try dragging whatever to the editor :smile: 
- [ ] The D-Bus interface. For example, try to open a custom menu exported from the menu editor.
- [ ] The example commands from the README.md
- [ ] Whatever else comes to our minds...

Once it is merged, I'll create a new issue for updating the Italian translation. Then  we can proceed to release version 4 as the change log is quite substantial already. The remaining issues can be rescheduled to Fly-Pie 5.

**Review by daPhipz — CHANGES_REQUESTED** [review]
Wow, this review turned out much bigger than expected :joy:
README is free of typos as far as I could see :+1: 
I spotted some inconsistencies between the original strings and the german translation, please have a look at those. Where applicable, I commented in the original source code and not in the `de.po` file.

Please, do not feel offended by my somewhat cold, short comments - I simply wanted to be efficient and quick. I have friendly intentions and by no means want to criticize your awesome work! Sorry if some comments seem harsh - they are by no means intended to insult or be rude! :kissing_heart:

File: po/de.po [hunk 1]
```diff
@@ -1671,84 +1400,95 @@ msgstr "Importieren der Menükonfiguration ist fehlgeschlagen!"
 msgid "Menu Structure"
 msgstr "Menüstruktur"
 
-#: settings/MenuEditor.js:1073
+#: settings/MenuEditor.js:1011 settings/MenuEditor.js:1040
+#: settings/MenuEditor.js:1261 settings/MenuEditor.js:1280
+msgid "Not bound."
+msgstr "Deaktiviert."
```
[the remark this answers is not in the document]
Comment (Schneegans) [reply po/de.po:1407]: I added a comment!

File: po/de.po [hunk 2]
```diff
@@ -744,368 +753,76 @@ msgstr ""
...
 msgstr "Dieser Textwird eingefügt."
 
-#: settings/settings.ui:7937
+#: settings/settings.ui:7932 settings/settings.ui:7968
+msgid "ID"
+msgstr "ID"
+
+#: settings/settings.ui:7945
+msgid "This will be passed to the D-Bus signal."
+msgstr "Diese ID wird dem D-Bus Signal übergeben."
```
[the remark this answers is not in the document]
Comment (Schneegans) [reply po/de.po]: I think it's ok without. I changed it accordingly.

File: po/de.po [hunk 3]
```diff
@@ -1769,3 +1509,292 @@ msgstr "<big>Letzte Auswahlzeit: <b>%d ms</b></big>"
...
+"However, for precise item selection, a maximum number of twelve items is "
+"recommended.\n"
+"Top-level menus can be opened using a shortcut. It is also possible to open "
+"a menu with a terminal command. You can read more on <a href=\"https://"
+"github.com/Schneegans/Fly-Pie\">Github</a>."
+msgstr ""
+"Ein <b>Benutzerdefiniertes Menü</b> kann beliebig viele Aktionen und "
+"Untermenüs enthalten. Für eine präzise Auswahl sind allerdings eine maximal "
+"Anzahl von zwölf Einträgen pro Ebene empfehlenswert.\n"
+"Das Menü kann mit obigem Tastaturkürzel geöffnet werden. Es ist auch "
```
[the remark this answers is not in the document]
Comment (Schneegans) [reply po/de.po]: I changed the entire description a little bit to make things clearer, I hope :smile:

File: po/de.po [hunk 4]
```diff
@@ -1769,3 +1509,292 @@ msgstr "<big>Letzte Auswahlzeit: <b>%d ms</b></big>"
...
+msgid ""
+"The <b>Favorites</b> menu shows the applications you have pinned to Gnome "
+"Shell's Dash."
+msgstr ""
+"Das <b>Favoriten</b>-Menü zeigt die Anwendungen, die Sie an die Dash von "
+"GNOME Shell angeheftet haben."
+
+#: common/menus/FrequentlyUsed.js:36
+msgid "Frequently Used"
+msgstr "Häufig verwendete Anwendungen"
```
[the remark this answers is not in the document]
Comment (Schneegans) [reply po/de.po]: Done.

File: po/de.po [hunk 5]
```diff
@@ -1769,3 +1509,292 @@ msgstr "<big>Letzte Auswahlzeit: <b>%d ms</b></big>"
...
+msgid "Shows your frequently used applications."
+msgstr "Zeigt oft verwendete Programme."
+
+#: common/menus/FrequentlyUsed.js:41
+msgid ""
+"The <b>Frequently Used</b> menu shows a list of frequently used "
+"applications. You should limit the maximum number of shown applications to a "
+"reasonable number."
+msgstr ""
+"Das <b>Häufig-verwendete-Anwendungen</b>-Menü zeigt eine Liste oft "
```
[the remark this answers is not in the document]
Comment (Schneegans) [reply po/de.po]: Done.

File: po/de.po [hunk 6]
```diff
@@ -1769,3 +1509,292 @@ msgstr "<big>Letzte Auswahlzeit: <b>%d ms</b></big>"
...
+msgstr "Zeigt kürzlich geöffnete Dateien."
+
+#: common/menus/RecentFiles.js:31
+msgid ""
+"The <b>Recent Files</b> menu shows a list of recently used files. You should "
+"limit the maximum number of shown files to a reasonable number."
+msgstr ""
+"Das <b>Zuletzt-verwendete-Dateien</b>-Menü zeigt eine Liste aller kürzlich "
+"verwendeten Dateien. Sie sollten die maximal angezeigte Anzahl auf etwa "
+"zwölf beschränken."
```
[the remark this answers is not in the document]
Comment (Schneegans) [reply po/de.po]: I changed the original string.

**Comment by Schneegans** [comment]
Thank you so much! This was really a lot!

**Comment by daPhipz** [comment]
No problem! Happy to help :)

**Comment by Schneegans** [comment]
I haven't spotted any bugs yet. So I will merge this now and create an issue for the Italian translation. If we discover any new bugs, we can create a new issue accordingly!

**Pull request merged**